dask/array/tests/test_array_core.py

```diff
@@ -2452,22 +2467,35 @@ def test_Array_normalizes_dtype():
     assert isinstance(x.dtype, np.dtype)
 
 
-def test_from_array_with_lock():
+@pytest.mark.parametrize("inline_array", [True, False])
+def test_from_array_with_lock(inline_array):
     x = np.arange(10)
-    d = da.from_array(x, chunks=5, lock=True)
 
-    tasks = [v for k, v in d.dask.items() if k[0] == d.name]
+    class FussyLock(SerializableLock):
+        def acquire(self, blocking=True, timeout=-1):
+            if self.locked():
+                raise RuntimeError("I am locked")
+            return super().acquire(blocking, timeout)
+
+    lock = FussyLock()
+    d = da.from_array(x, chunks=5, lock=lock, inline_array=inline_array)
 
-    assert hasattr(tasks[0][4], "acquire")
-    assert len({task[4] for task in tasks}) == 1
+    with pytest.raises(RuntimeError):
+        lock.acquire()
```

Comment:
```suggestion
    lock.acquire()
    with pytest.raises(RuntimeError):
```
I believe this first `acquire` should not raise?

Reply:
:+1: